docs/DATA_FLOW.md

```diff
@@ -0,0 +1,629 @@
+    
+    FQ -->|Cordon Node| K8S
+    ND -->|Drain Pods| K8S
+    FR -->|Create CRD| K8S
+    LBL -->|Update Labels| K8S
+    
+    style DB fill:#f9f,stroke:#333,stroke-width:2px
+    style PC fill:#bbf,stroke:#333,stroke-width:2px
+```
+
+---
+
+## Core Data Structure: HealthEvent
+
+All data flowing through NVSentinel is based on the `HealthEvent` protobuf message:
+
+### HealthEvent Message Structure
+
+```protobuf
+message HealthEvent {
+  uint32 version = 1;                          // Protocol version
+  
+  // Source identification
+  string agent = 2;                            // Monitor name (e.g., "gpu-health-monitor")
+  string componentClass = 3;                   // Component type (e.g., "GPU", "SYSLOG")
+  string nodeName = 13;                        // Kubernetes node name
+  
+  // Health status
+  string checkName = 4;                        // Specific check (e.g., "XID_ERROR")
+  bool isFatal = 5;                           // Critical failure
+  bool isHealthy = 6;                         // Current health state
+  string message = 7;                         // Human-readable description
+  
+  // Classification
+  RecommendedAction recommendedAction = 8;    // What should be done
+  repeated string errorCode = 9;              // Error identifiers (e.g., ["XID-48"])
+  repeated Entity entitiesImpacted = 10;      // Affected resources
+  
+  // Metadata
+  map<string, string> metadata = 11;          // Key-value pairs (GPU UUID, driver version, etc.)
+  google.protobuf.Timestamp generatedTimestamp = 12;
+  
+  // Behavior overrides
+  BehaviourOverrides quarantineOverrides = 14;
+  BehaviourOverrides drainOverrides = 15;
+}
+
+enum RecommendedAction {
+  NONE = 0;
+  COMPONENT_RESET = 2;
+  CONTACT_SUPPORT = 5;
+  RESTART_VM = 15;
+  RESTART_BM = 24;
+  REPLACE_VM = 25;
+  UNKNOWN = 99;
+}
+
+message Entity {
+  string entityType = 1;    // e.g., "GPU", "NODE", "POD"
+  string entityValue = 2;   // e.g., GPU UUID, node name
+}
+```
+
+### Example HealthEvent: GPU XID Error
+
+```json
+{
+  "version": 1,
+  "agent": "gpu-health-monitor",
+  "componentClass": "GPU",
+  "checkName": "XID_ERROR_48",
+  "isFatal": true,
+  "isHealthy": false,
+  "message": "GPU 0 reported XID 48 (Double Bit ECC Error)",
+  "recommendedAction": "REPLACE_VM",
+  "errorCode": ["XID-48"],
+  "entitiesImpacted": [
+    {
+      "entityType": "GPU",
+      "entityValue": "GPU-12345678-abcd-1234-abcd-123456789abc"
+    }
+  ],
+  "metadata": {
+    "gpu_uuid": "GPU-12345678-abcd-1234-abcd-123456789abc",
+    "gpu_index": "0",
+    "driver_version": "535.104.05",
+    "severity": "CRITICAL"
+  },
+  "generatedTimestamp": "2025-10-28T10:15:30Z",
+  "nodeName": "gpu-node-42",
+  "quarantineOverrides": null,
+  "drainOverrides": null
+}
+```
+
+---
+
+## Component Data Flow
+
+### 1. GPU Health Monitor
+
+**What it captures:**
+- GPU temperature, power, utilization (via DCGM)
+- ECC errors (single-bit, double-bit)
+- XID errors (hardware faults)
```

Comment:
XID error are detected only in the syslog health monitor and not in GPU healthy monitor anymore